query-coordinator/src/main/scala/com/socrata/querycoordinator/QueryParser.scala

```diff
@@ -24,11 +25,25 @@ class QueryParser(analyzer: SoQLAnalyzer[SoQLType, SoQLValue], schemaFetcher: Sc
 
   type AnalysisContext = com.socrata.soql.AnalysisContext[SoQLType, SoQLValue]
 
-  private def go(columnIdMapping: Map[ColumnName, String], schema: Map[String, SoQLType])
+  private def go(columnIdMapping: Map[ColumnName, String], schema: Map[String, SoQLType], context: Context, lensUid: Option[String])
                 (f: AnalysisContext => (BinaryTree[SoQLAnalysis[ColumnName, SoQLType]], Map[QualifiedColumnName, String], DateTime)): Result = {
+      val parameters = (context.user.num.toSeq ++
+        context.user.text.toSeq ++
+        context.user.bool.toSeq ++
+        context.user.fixed.toSeq ++
+        context.user.floating.toSeq)
+        .filter { case(name,_) => name.contains(".")}
 
-    val ds = toAnalysisContext(dsContext(columnIdMapping, schema))
     try {
+      val uid = lensUid match {
+        case Some(uid) => uid
+        case None => throw new ParameterSpecException("lensUid missing from request. It is needed to build the ParamterSpec")
```

Comment:
maybe we only want to do this if there `parameters` is non-empty, otherwise just use `ParameterSpec.empty`

Reply:
yeah i think this is better